src/solver/tensorOp/Op2dTensorLite.cpp

```diff
@@ -0,0 +1,198 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include "tensor_op_helpers.hpp"
+
+#include <miopen/tensorOp/solvers.hpp>
+#include <miopen/tensorOp/invoke_params.hpp>
+#include <miopen/tensor.hpp>
+#include <miopen/kernel_build_params.hpp>
+#include <miopen/float_equal.hpp>
+#include <miopen/datatype.hpp>
+
+namespace miopen {
+
+namespace solver {
+
+namespace tensorOp {
+
+bool Op2dTensorLite::IsApplicable(const ExecutionContext& context,
+                                  const miopen::tensorOp::ProblemDescription& problem) const
+{
+    const auto& aTensorDesc = problem.GetATensorDesc();
+    const auto& bTensorDesc = problem.GetBTensorDesc();
+    const auto& cTensorDesc = problem.GetCTensorDesc();
+
+    const auto& alens = aTensorDesc.GetLengths();
+    const auto& blens = bTensorDesc.GetLengths();
+    const auto& clens = cTensorDesc.GetLengths();
+
+    auto asize = alens.size();
+
+    if(GetDataType(aTensorDesc.GetType()) == "double")
+    {
+        return false;
+    }
+
+    if(asize < 3)
+    {
+        return false;
+    }
```

Comment:
It can be applicable in 2d case, it forces it explicitly in this statement `clens[0] == 1 && blens[0] == 1 && alens[0] == 1`.
But it introduces some complications later.

Probably that `isApplicable` tuning can be done in a separate PR.

Also it's not a `asize < 3`, it must be `asize != 3`. That should be fixed in this PR, because if `asize == 4`, it'll break everything in `GetSolution`

Reply:
I think that it is better to keep `isApplicable` straightforward and same applications as they are in `tensorocl.cpp` file. That additional tuning can be done later in separate PR, I can also open an issue for that to be completed later.